File: Examples/CSharp/Articles/RenderingAndPrinting/ExportChartToSvgWithViewBox.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Aspose.Cells.Examples.CSharp.Articles.RenderingAndPrinting
{
    public class ExportChartToSvgWithViewBox
    {
        public static void Run()
        {
            //Source directory
            string sourceDir = RunExamples.Get_SourceDirectory();

            //Output directory
            string outputDir = RunExamples.Get_OutputDirectory();

            // Create workbook object from source file
            Workbook workbook = new Workbook(sourceDir + "sampleExportChartToSvgWithViewBox.xlsx");

            // Access first worksheet
            Worksheet worksheet = workbook.Worksheets[0];

            // Access first chart inside the worksheet
            Aspose.Cells.Charts.Chart chart = worksheet.Charts[0];

            // Set image or print options with SVGFitToViewPort true
            Aspose.Cells.Rendering.ImageOrPrintOptions opts = new Aspose.Cells.Rendering.ImageOrPrintOptions();
            opts.SaveFormat = SaveFormat.Svg;
            opts.SVGFitToViewPort = true;

            // Save the chart to svg format
            chart.ToImage(outputDir + "outputExportChartToSvgWithViewBox.svg", opts);

            Console.WriteLine("ExportChartToSvgWithViewBox executed successfully.");
        }
    }
}

File: Examples/VisualBasic/Articles/RenderingAndPrinting/ExportChartToSvgWithViewBox.vb

Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace Articles.RenderingAndPrinting
    Public Class ExportChartToSvgWithViewBox
        Public Shared Sub Run()
            ' ExStart:ExportChartToSvgWithViewBox
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

            ' Create workbook object from source file
            Dim workbook As New Workbook(dataDir & Convert.ToString("SampleChartBook.xlsx"))

            ' Access first worksheet
            Dim worksheet As Worksheet = workbook.Worksheets(0)

            ' Access first chart inside the worksheet
            Dim chart As Aspose.Cells.Charts.Chart = worksheet.Charts(0)

            ' Set image or print options with SVGFitToViewPort true
            Dim opts As New Aspose.Cells.Rendering.ImageOrPrintOptions()
            opts.SaveFormat = SaveFormat.SVG
            opts.SVGFitToViewPort = True

            ' Save the chart to svg format
            chart.ToImage(dataDir & Convert.ToString("Image_out_.svg"), opts)
            ' ExEnd:ExportChartToSvgWithViewBox
        End Sub
    End Class
End Namespace
